Refine ignore terminal.

Waitzz opened the pull request:
There is a error  highlight on terminal, reproduction steps :
1. cd ~/.local/share/lazy/whitespace.nvim
2. nvim
3. :e lua/whitespace-nvim/init.lua  (in nvim's cmdline)  have a correct highlight
4. :terminal (in nivm's cmdline) have a error highlight

terminal buffer will not set fietype so the "FileType" autocmd will not be excuted , I add a "TermOpen" autocmd to clean match if "ignore_terminal" is set to true, add clean function for user can clean match if they want. and I think we should need put any conditions in hightlignt function , they should be put in function who call them.

please confirm for me , best wishes.

Commit:
Refine ignore terminal.

Waitzz commented:
my test nvim version infomations : nvim --version
NVIM v0.10.0
Build type: Release
LuaJIT 2.1.1713484068
Run "nvim -V1 -v" for more info

johnfrankmorgan commented on lua/whitespace-nvim/init.lua:
```diff
@@ -36,15 +30,28 @@ whitespace.trim = function()
 end
 
+local function file_type()
+  if vim.tbl_contains(config.ignored_filetypes, vim.bo.filetype) then
+    whitespace.clean()
+    return
+  end
+  whitespace.highlight()
```
Can you use `if ... else` here like in `term_open` please? That way it's more consistent

johnfrankmorgan commented on lua/whitespace-nvim/init.lua:
```diff
@@ -7,21 +7,15 @@ local config = {
 
 local whitespace = {}
 
+whitespace.clean = function()
```
Maybe this could be called `nohighlight` or `remove_highlight`?

johnfrankmorgan commented:
Thanks for the PR @Waitzz, I've left a few comments. Let me know what you think.

Waitzz committed:
Rename clean to nohighlight and refine file_type()

Waitzz commented:
I have modified it according to your suggestion, please review it again, thank you ~

Commit:
Fix wrong highlight when switching between terminal and code file

reproduction steps:
1. type :e to edit a lua file or other file not be ignored
2. type :terminal to open a terminal buffer
3. type :e to previous lua file or other file not be ignored
4. wrong highlight happened

Change-Id: Ib96951e47f16f75cd7a7a7a84f1a9125e6a20c1e

johnfrankmorgan commented:
Thanks @Waitzz!
